app/models/job_preferences/job_scope.rb

```diff
@@ -24,27 +18,11 @@ def call
 
     attr_reader :scope, :job_preferences
 
-    # TODO: Delete whole mapping and directly query by "job_preference.roles" once there are no more vacancies using
-    # "senior_leader" and "middle_leader" roles.
-    #
-    # Vacancies at the moment still use the generic "senior_leader" and "middle_leader" roles.
-    # Jobseeker preferences split those roles into more granular roles:
-    # - "headteacher" => "headteacher", "deputy_headteacher", "assistant_headteacher"
-    # - "middle_leader" => "head_of_year_or_phase", "head_of_department_or_curriculum"
-    #
-    # This method ensures that vacancies using the old generic roles are matched with jobseekers with granular roles
-    #
-    # If any of the granular roles is present in the Jobseeker preferences, we add the generic role in the vacancy search.
-    #
-    # EG:
-    # - Jobseeker role preferences: ["headteacher", "assistant_headteacher"]
-    # - Method output: ["headteacher", "assistant_headteacher", "senior_leader"]
-    # - Vacancies with role "senior_leader" will be returned.
-    # - When vacancies are migrated to the new granular roles, vacancies wih roles "headteacher" and/or "assistant_headteacher"
-    #   will be returned.
-    def roles
-      job_preferences.roles.tap do |roles|
-        ROLE_MAPPINGS.each { |generic_role, granular_roles| roles << generic_role if granular_roles.intersect?(roles) }
+    def apply_job_roles(scope)
```

Comment:
`with_any_of_job_roles` is a scope method provided by `array_enum` gem. It raises an exception if there is no values passed (`[]`) to th e call. So had to add this `if/else`  to ensure we cover both situations while returning a valid scope.